snuba_sdk/request.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import json
+import re
+from dataclasses import dataclass, fields
+
+from snuba_sdk.query import Query
+
+
+class InvalidRequestError(Exception):
+    pass
+
+
+class InvalidFlagError(Exception):
+    pass
+
+
+@dataclass(frozen=True)
+class Flags:
+    totals: bool | None = None
+    consistent: bool | None = None
+    turbo: bool | None = None
+    debug: bool | None = None
+    dry_run: bool | None = None
+    legacy: bool | None = None
+
+    def validate(self) -> None:
+        flags_fields = fields(self)
+        for ff in flags_fields:
+            flag = getattr(self, ff.name)
+            if flag is not None and not isinstance(flag, bool):
+                raise InvalidFlagError(f"{ff.name} must be a boolean")
+
+    def to_dict(self) -> dict[str, bool]:
+        self.validate()
+        values = {
+            "totals": self.totals,
+            "consistent": self.consistent,
+            "turbo": self.turbo,
+            "debug": self.debug,
+            "dry_run": self.dry_run,
+            "legacy": self.legacy,
+        }
+        return {f: v for f, v in values.items() if v is not None}
+
+
+FLAG_RE = re.compile(r"^[a-zA-Z0-9_\.\+\*\/:\-\[\]]*$")
+
+
+@dataclass
+class Request:
+    dataset: str
```

Comment:
It isn't part of the SnQL syntax. There is a little bit of legacy stuff going on here. The dataset is separate since the URL has the dataset in it, so it shouldn't be encoded in the Query. I think the definition of a dataset needs to evolve a bit.